src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,46 +509,190 @@ public int getID()
     }
 
     /**
-     * Increment the mobs killed by this colony.
-     * <p>
-     * Will award achievements for mobs killed.
+     * Increment the statistic amount and trigger achievement.
+     * @param statistic the statistic.
      */
-    public void incrementMobsKilled()
+    public void incrementStatistic(@NotNull String statistic)
     {
-        killedMobs++;
-        final int mobKills = this.getKilledMobs();
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIRST)
+        final int statisticAmount = this.getStatisticAmount(statistic);
+        switch (statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)